src/Servers/Kestrel/Core/src/Internal/Http2/Http2FrameWriter.cs

```diff
@@ -566,32 +572,107 @@ private ValueTask<FlushResult> WriteDataAndTrailersAsync(Http2Stream stream, in
         }
     }
 
-    private void FinishWritingHeadersUnsynchronized(int streamId, int payloadLength, bool done)
+    private void SplitHeaderFramesToOutput(int streamId, ReadOnlySpan<byte> dataToFrame, bool endOfHeaders, bool isFramePrepared)
```

Comment:
Personally, I might call this something like "SplitHeaderAcrossFrames" or "WriteHeaderAsMultipleFrames".  Basically, I think the name should convey that a single header will be written in multiple parts (assuming I've understood the method correctly).

Reply:
I think there are cases where this method is called with header data that doesn't need to be split.

Reply:
Yes, such can case exists with the regular "MoreHeaders" result. I don't have a better name in mind, so I would go with the suggested `SplitHeaderAcrossFrames`.

Reply:
IMO: WriteHeaderDataAcrossFrames + comment that header data is written across 1-to-n frames, based on the header data size and max frame size.